src/box/memtx_engine.cc

```diff
@@ -2185,25 +2186,7 @@ memtx_index_extent_reserve(struct memtx_engine *memtx, int num)
 			 "mempool", "new slab");
 		return -1;
 	});
-	struct mempool *pool = &memtx->index_extent_pool;
-	while (memtx->num_reserved_extents < num) {
-		void *ext;
-		while ((ext = mempool_alloc(pool)) == NULL) {
-			bool stop;
-			memtx_engine_run_gc(memtx, &stop);
-			if (stop)
-				break;
-		}
-		if (ext == NULL) {
-			diag_set(OutOfMemory, MEMTX_EXTENT_SIZE,
-				 "mempool", "new slab");
-			return -1;
-		}
-		*(void **)ext = memtx->reserved_extents;
-		memtx->reserved_extents = ext;
-		memtx->num_reserved_extents++;
-	}
-	return 0;
+	return matras_allocator_reserve(&memtx->index_extent_allocator, num);
```

Comment:
The patch looks good to me, but I'd like to see block reservation done in indexes (rtree, bps, light) to make sure the new matras allocator API fits. Let's try it if it doesn't take too long.

Reply:
Discussed f2f, it's been decided to finish with RTREE in the future: #10831.